tests/test_custom_objects_rest.py

```diff
@@ -0,0 +1,104 @@
+"""
+Test that all fields can be replicated for a stream that is a custom object (REST API)
+"""
+from datetime import datetime, timedelta
+
+from tap_tester import connections, menagerie, runner
+
+from sfbase import SFBaseTest
+
+
+class SalesforceCustomObjects(SFBaseTest):
+    """Test that all fields can be replicated for a stream that is a custom object"""
+
+    # Note: custom record as seen Aug 9, 2023
+    # TODO do we need to hard code start date to 08-04?
+    # 'CreatedDate': '2023-08-04T20:13:50.000000Z'
+    # 'SystemModstamp': '2023-08-04T20:13:50.000000Z',
+    # 'LastModifiedDate': '2023-08-04T20:13:50.000000Z',
+    # 'LastReferencedDate': '2023-08-09T20:19:45.000000Z',
+    # 'LastViewedDate': '2023-08-09T20:19:45.000000Z',
+    start_date = (datetime.now() + timedelta(days=-1)).strftime("%Y-%m-%dT00:00:00Z")
+
+    @staticmethod
+    def expected_sync_streams():
+        return {
+            'TapTester__c',  # 1 created record present, spike on CRUD?
+            # 'TapTester__Share',  # TODO no data, fix or omit?
+        }
+
+    def custom_objects_test(self):
+        """
+        Verify that all fields can be replicated for a custom  stream
+
+        PREREQUISITE
+        Create a custom object and at least one data record for that cusotm object
+        """
+
+        expected_streams = self.expected_sync_streams()
+
+        # instantiate connection
+        conn_id = connections.ensure_connection(self)
+
+        # run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        test_catalogs = [catalog for catalog in found_catalogs
+                                          if catalog.get('stream_name') in expected_streams]
```

Comment:
Style Note: The second line of a longer line usually lines up with second character of the first line.

```python
result = [{'key': value} for value in iterable
          if a_long_filter_expression(value)]
```
[Source](https://google.github.io/styleguide/pyguide.html#274-decision)

---

My personal list comprehension rule is that if it doesn't fit on one line, each segment gets its own line

```python
test_catalogs = [
    catalog 
    for catalog in found_catalogs
    if catalog.get('stream_name') in expected_streams
]
```

Reply:
Updated